build/lib/src/builder/build_step_impl.dart

```diff
@@ -57,11 +57,18 @@ class BuildStepImpl implements BuildStep {
 
   bool _isComplete = false;
 
+  @override
+  final removedDependencies = <AssetId>{};
+
   BuildStepImpl(this.inputId, Iterable<AssetId> expectedOutputs, this._reader,
       this._writer, this._rootPackage, this._resolvers, this._resourceManager,
       [this._stageTracker = NoOpStageTracker.instance])
       : _expectedOutputs = expectedOutputs.toSet();
 
+  @override
+  void removeDependency(AssetId id) =>
+      (removedDependencies as Set<AssetId>).add(id);
```

Comment:
These casts are unfortunate, and I end up doing it in build_runner_core right now also - so this is a place where the current api is not ideal for sure.

Reply:
I can guard the cast with an is check and then do `toSet` if it fails.

Reply:
nvm this no longer exists